File: 01_StudentCode/01_A/3A/computer_science_exercises/02_controlFlow/controlFlow.py
# Control Flow, ryan Kelley, v0.1 

# DECLARATIONS 
favColor = "Blue"
luckyNumber = 25
myGPA = 3.0
myAge = 66 
pineappleOnPizza = True 

# if Statement 
if luckyNumber > 30: # luckyNumber > 30 is a CONDITIONAL EXPRESSION 
    print("Wow, what a great number!")

if favColor != "Green":
    print("Your favorite color stinks.")

# if-else Statement  
if myGPA >= 3.0: 
    print("Congratulations on making the honor roll!")
else: 
    print("You did not make the honor roll.")

# if-elif-else Statement 
# if myAge > 65: 
#     print("Please enjoy retirement!")
# elif myAge > 55: 
#     print("You have earned a $10,000 bonus!")
# elif myAge > 40: 
#     print("You have earned a $5,000 bonus.")
# else:
#     print("You have earned a $1,000 bonus.")
# 99% OF THE TIME CHECK FOR HIGHEST VALUES FIRST! 

if myAge > 40: 
    print("You have earned a $5,000 bonus.")
elif myAge > 55: 
    print("You have earned a $10,000 bonus!")
elif myAge > 65: 
    print("Please enjoy retirement!")    
else:
    print("You have earned a $1,000 bonus.")

# NESTED IF / ELIF / ELSE STATEMENTS    
if pineappleOnPizza == True: 
    print("You have strange tastes.  I bet you're old.")
    if myAge > 50:
        print("What was it like riding to school on a dinosaur?")                
    else: 
        print("Ok, you're still young enough to be cool.")
else:
    print("Ok good, you eat pizza like a normal human.")    
     
# while Loop -- Think MUSICAL CHAIRS -- Best used when you DO NOT know how many times the loop must run. 
# loopCounter = 0 
# while loopCounter <= 100: 
#     print(f"The current loop count is {loopCounter}.")
#     loopCounter += 1
# # loopCounter is equal to 99 after finishing! 
# while loopCounter > 0: 
#     print(f"The current loop count is {loopCounter}.")
#     loopCounter -= 1 

loopCounter = 0 
# while loopCounter >= 0: 
#     print(f"The current loop count is {loopCounter}.")
#     if loopCounter % 2 == 0:
#         print(f"{loopCounter} is even!")
#     else: 
#         print(f"{loopCounter} is odd!")
#     loopCounter += 1

# for Loop -- Think Go Fish!  Used when you know number of iterations needed.
for eachNumber in range(10): 
    # eachNumber is known as the 'iterable variable' 
    # range(x) specifies the numbers from 0 to x, NOT INCLUSIVE 
    print(eachNumber) 

File: 01_StudentCode/01_A/4A/computer_science_exercises/02_controlFlow/controlFlow.cs
// Ryan Kelley, Program Template, v0.01
using System; 

namespace controlFlow 
{
    class controlFlow 
    {
        static void Main(string[] args)
        {
            // DECLARATIONS 
            string favColor = "Blue"; 
            int luckyNumber = -5; 
            float myGPA = 1.33F;
            int myAge = 43; 
            bool pineappleOnPizza = true; 
            /*
            // if Statements -- Check for a single condition!
            if (favColor == "Green") {
                Console.WriteLine("Green with envy!");
            }

            if (luckyNumber % 2 == 0 ) {
                Console.WriteLine("Your lucky number is even!");
            }

            // if - else Statement -- Checks for a single condition, but has two actions. 
            if (myAge > 15) {
                Console.WriteLine("You are eligible to drive");                
            } else {
                Console.WriteLine("START WALKING SCRUB!");
            }

            // if -- else if -- else -- Checks multiple outcomes. 
            if (myGPA == 4.0F) {
                Console.WriteLine("Congrats on straight A grades!");                
            } else if (myGPA >= 3.0F) {
                Console.WriteLine("Congrats on making honor roll!");                
            } else if (myGPA >= 2.0F) {
                Console.WriteLine("You are graduation ready!");                
            } else {
                Console.WriteLine("You should probably study!");                
            }
            // WHEN CHECKING NUMBER VALUES, START WITH THE HIGHEST VALUE!!!!

            if (myGPA >= 2.0F) {
                Console.WriteLine("You are graduation ready!");                     
            } else if (myGPA >= 3.0F) {
                Console.WriteLine("Congrats on making honor roll!");                
            } else if (myGPA == 4.0F) {
                Console.WriteLine("Congrats on straight A grades!");                           
            } else {
                Console.WriteLine("You should probably study!");                
            }

            // Nested Statements 
            if (pineappleOnPizza == true) {
                Console.WriteLine("Eww, that's gross. You must be a boomer.  How old are you?");
                if (myAge > 60) {
                    Console.WriteLine("Just as I suspected! What was it like having a dinosaur growing up?");
                } else {
                    Console.WriteLine("Ok, so you're not a boomer but still have no taste in food.");
                }
            } else {
                Console.WriteLine("Glad to see you have common sense!"); 
            }
            */ 
            // for Loop -- Best for when you know # of iterations needed.
            /* 
                for (statement1; statement2; statement3) {
                Code to loop.
            }
            statement1 is executed ONCE BEFORE the loop starts. 
            statement2 is a CONDITIONAL that is checked EVERYTIME BEFORE loop starts. 
            statement3 is executed EVERYTIME after the loop executes. 
            */ 
            /*
            for (int i = 0; i < 101; i++) {
                Console.WriteLine("" + i);
            }

            // Create your own loop that counts down from 100 to 0. 
            for (int i = 100; i > -1; i--) {
                Console.WriteLine("" + i);
            }
            
            // Nested loops 
            // Outer Loop
            for (int i = 1; i <= 2; i++) {
                Console.WriteLine("Outer: " + i); 

                for (int j = 1; j <= 3; j++) {
                    Console.WriteLine("Inner: " + j);
                }
            }

            // while loop -- Best used when # of iterations needed is unknown
            int x = 0; 
            while (x < 1000) {
                Console.WriteLine("" + x);
                x++; 
            }
            
            // Special Keywords 
            // break will immediately exit a LOOP or an IF/ELSE IF/ELSE block.
            for (int i = 0; i < 101; i++) {
                Console.WriteLine("" + i);
                if (i == 50) {
                    break;
                }
            }
            */ 
            // continue will SKIP the current iteration and then finish the loop.
            for (int i = 0; i < 101; i++) {
                Console.WriteLine("" + i);
                if (i == 50) {
                    continue;
                }
            }

        }
    }
}
